sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/BenchmarkRequestSubscriber.java

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.benchmark;
+
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.Timer;
+import org.reactivestreams.Subscription;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.BaseSubscriber;
+
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class BenchmarkRequestSubscriber<T> extends BaseSubscriber<T> {
+    final Logger logger;
+    private Meter successMeter;
+    private Meter failureMeter;
+    private Semaphore concurrencyControlSemaphore;
+    private  AtomicLong count;
+    Timer.Context context;
+
+    public BenchmarkRequestSubscriber(Meter successMeter, Meter failureMeter, Semaphore concurrencyControlSemaphore,  AtomicLong count) {
+        this.successMeter = successMeter;
+        this.failureMeter = failureMeter;
+        this.concurrencyControlSemaphore = concurrencyControlSemaphore;
+        this.count = count;
+        logger = LoggerFactory.getLogger(this.getClass());
```

Comment:
this class is getting instantiated in a for loop in your benchmark, meaning the logger initialization will be called per for loop iteration.

logger should be static to avoid initialization cost per loop iteration.

Reply:
done